backend/test/modules/match-module/presentation/rest-api/MatchRestApiModule.spec.ts

```diff
@@ -118,4 +119,50 @@ describe('Match REST API', () => {
     expect(status).toBe(StatusCodes.NOT_FOUND);
     expect(body).toStrictEqual({ message: 'Match with id = sampleMatchId was not found!' });
   });
+
+  it('POST /rest-api/matches/:matchId/result | when command end match success', async () => {
+    //Given
+    const commandPublisher = CommandPublisherMock(CommandResult.success());
+    const { agent } = testModuleRestApi(MatchRestApiModule, { commandPublisher });
+
+    //When
+    const { body, status } = await agent
+      .post('/rest-api/matches/sampleMatchId/result')
+      .send({ matchId: 'sampleMatchId', winnerId: 'team1Id' });
+
+    //Then
+    expect(commandPublisher.executeCalls).toBeCalledWith(new EndMatch({ matchId: 'sampleMatchId', winnerId: 'team1Id' }));
+    expect(status).toBe(StatusCodes.OK);
+    expect(body).toBeEmpty();
+  });
+
+  it('POST /rest-api/matches/:matchId/result | when command end match fails due to wrong id given', async () => {
+    //Given
+    const commandPublisher = CommandPublisherMock(CommandResult.failureDueTo(new Error("Cannot end match that hasn't started.")));
+    const { agent } = testModuleRestApi(MatchRestApiModule, { commandPublisher });
+
+    //When
+    const { body, status } = await agent.post('/rest-api/matches/NotStartedMatchId/result').send({ winnerId: 'team1Id' });
+
+    //Then
+    expect(commandPublisher.executeCalls).toBeCalledWith(new EndMatch({ matchId: 'NotStartedMatchId', winnerId: 'team1Id' }));
+    expect(status).toBe(StatusCodes.BAD_REQUEST);
+    expect(body).toStrictEqual({ message: "Cannot end match that hasn't started." });
+  });
+
+  it('POST /rest-api/matches/:matchId/result | when command end match fails due to wrong winnerId given', async () => {
```

Comment:
Akurat na poziomie testów presentation (REST API) nie potrzeba rozgraniczać przypadków niepowodzenia. Ważne jest dla nas, że zostanie przekazane to co zostało zwrócone jako błąd z commandHandlera. 

Test tego przypadku powinniśmy przenieść do application, bo to rzeczywiście tam jest testowany ten kod. Teraz on nawet nie jest wywołany i nie sprawdzamy tego warunku.

Reply:
Ten przypadek jest przetestowany w application, z resztą jak każdy, w którym komenda może zaliczyć faila. To w takim razie
mam usunąć te test case'y w api, które sprawdzają faila czy pozmieniać ich nazwy?

Reply:
Jak uważasz :) Ja bym nie dodawał do presentation tych przypadków, bo to nic nowego nie testuje. Pokrycie jest takie samo.